### PR TITLE
Retry primary handshake 3 times instead of 100 in embedded replicas

### LINKED ISSUE #1676: Sync hangs for a very long time when using archived database
Having `Cargo.toml`:

```
[package]
name = "latencies-test"
version = "0.1.0"
edition = "2021"

[dependencies]
libsql = { git = "https://github.com/tursodatabase/libsql.git" }
tokio = { version = "1", features = ["full"] }
```

and `main.rs`:

```
use std::time::Instant;
use libsql::Builder;

#[tokio::main]
async fn main() {
    let url = "libsql://foo-ospfranco.turso.io";
    let token = "<token>";
      let db = Builder::new_remote_replica("test.db", url.into(), token.into())
          .build()
          .await
          .expect("failed to connect to db");

      let before = Instant::now();
      let conn = db.connect().expect("failed to connect to db");
      let after = Instant::now();
      println!("Time to connect: {:?}", after.duration_since(before));

      let before = Instant::now();
      let res = db.sync().await;
      let after = Instant::now();
      println!("Time to sync: {:?}", after.duration_since(before));
      res.expect("failed to sync db");

      let before = Instant::now();
      let _ = conn.query("select emp_no, first_name, last_name from employees limit 10", ()).await.expect("failed to query db");
      let after = Instant::now();
      println!("Time to query: {:?}", after.duration_since(before));
}
```

the program returns:
```
Time to connect: 1.128602ms
Time to sync: 1108.274159712s
thread 'main' panicked at src/main.rs:31:13:
failed to sync db: Replication(PrimaryHandshakeTimeout)
stack backtrace:
   0: rust_begin_unwind
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/std/src/panicking.rs:652:5
   1: core::panicking::panic_fmt
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/core/src/result.rs:1679:5
   3: core::result::Result<T,E>::expect
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/core/src/result.rs:1059:23
   4: latencies_test::main::{{closure}}
             at ./src/main.rs:31:9
   5: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/core/src/future/future.rs:123:9
   6: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/park.rs:281:63
   7: tokio::runtime::coop::with_budget
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/coop.rs:107:5
   8: tokio::runtime::coop::budget
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/coop.rs:73:5
   9: tokio::runtime::park::CachedParkThread::block_on
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/park.rs:281:31
  10: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/context/blocking.rs:66:9
  11: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/scheduler/multi_thread/mod.rs:87:13
  12: tokio::runtime::context::runtime::enter_runtime
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/context/runtime.rs:65:16
  13: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/scheduler/multi_thread/mod.rs:86:9
  14: tokio::runtime::runtime::Runtime::block_on_inner
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/runtime.rs:363:45
  15: tokio::runtime::runtime::Runtime::block_on
             at /Users/haaawk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/runtime.rs:333:13
  16: latencies_test::main
             at ./src/main.rs:10:5
  17: core::ops::function::FnOnce::call_once
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Timeouts must be misconfigured on the hello request I assume.

### PR DESCRIPTION
**Pull request by haaawk**
Fixes #1676

### CODE REVIEW

File: libsql-replication/src/replicator.rs [hunk 1]
```diff
@@ -208,7 +215,7 @@ where
 
     pub async fn try_perform_handshake(&mut self) -> Result<(), Error> {
```
Comment (sivukhin) [inline libsql-replication/src/replicator.rs:216]: Will it be helpful if we will expose metric about actual amount of attempts?

Comment (haaawk) [reply libsql-replication/src/replicator.rs:216]: Possibly but there are two cases here: embedded replicas and regular replicas. If we decide to add such metric it should be a separate PR anyways.

Comment (sivukhin) [reply libsql-replication/src/replicator.rs:216]: Ok...

File: libsql-replication/src/replicator.rs [hunk 2]
```diff
@@ -185,14 +186,20 @@ where
             injector,
             state: ReplicatorState::NeedHandshake,
             frames_synced: 0,
+            max_handshake_retries: HANDSHAKE_MAX_RETRIES,
```
[the remark this answers is not in the document]
Comment (haaawk) [reply libsql-replication/src/replicator.rs:189]: What difference does it make if the default is 100 and embedded replica sets the limit to 3 vs default being 3 and server setting the limit to 100?

**Pull request merged**